### PR TITLE
add TOC

### PR DESCRIPTION
**Pull request by namerhila**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

This PR fix this issue https://github.com/fastify/fastify/issues/3010

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -184,6 +184,7 @@
     "find-my-way": "^4.0.0",
     "flatstr": "^1.0.12",
     "light-my-request": "^4.2.0",
+    "markdown-it": "^12.0.6",
```
Comment (jsumners) [inline package.json:187]: Why have you added this new dependency?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,25 @@ Disclosure](https://img.shields.io/badge/Security-Responsible%20Disclosure-yello
...
+ - [Example](https://github.com/fastify/fastify#example)
+ - [Fastify v1.x and v2.x](https://github.com/fastify/fastify#fastify-v1x-and-v2x)
+ - [Core features](https://github.com/fastify/fastify#core-features)
+ - [Benchmarks](https://github.com/fastify/fastify#benchmarks)
+ - [Documentation](https://github.com/fastify/fastify#documentation)
+ - [Ecosystem](https://github.com/fastify/fastify#ecosystem)
+ - [Support](https://github.com/fastify/fastify#support)
+ - [Team](https://github.com/fastify/fastify#team)
+ - [Hosted by](https://github.com/fastify/fastify#hosted-by)
+ - [License](https://github.com/fastify/fastify#license)
```
Comment (jsumners) [inline README.md:40]: These should be relative links.

**Review by jsumners — CHANGES_REQUESTED** [review]

**Review by RafaelGSS — CHANGES_REQUESTED** [review]
The new feature from GitHub doesn't solve it? 
image

**Comment by RafaelGSS** [comment]
> The new feature from GitHub doesn't solve it?

I've just seen that we parse the table of contents on our website.

**Review by Eomm — COMMENTED** [review]
> This PR fix this issue #3010

Nope, that issue refers to this page:

image

**Review by mcollina — APPROVED** [review]
lgtm with @jsumners changes

**Pull request closed without merging**